### ISSUE TITLE
Crashes when exporting as MP3

### ISSUE THREAD

**Issue by Follery** [HielkeMinecraft/OpenNoteBlockStudio#86]
A song I am trying to export as MP3 does not work. The program crashes, and an empty .mp3 file is created instead of the actual song. The error code is as follows:

```
___________________________________________
############################################################################################
FATAL ERROR in
action number 1
of Draw Event
for object obj_controller:

Push :: Execution Error - Variable Index [0,14] out of range [1,14] - -1.lockedlayer(100211,14)
at gml_Script_mp3_export
############################################################################################
--------------------------------------------------------------------------------------------
stack frame is
gml_Script_mp3_export (line -1)
gml_Script_draw_window_mp3_export
gml_Script_draw_windows
gml_Script_control_draw
gml_Object_obj_controller_Draw_0
```

[The song I am trying to export is linked here.](https://drive.google.com/open?id=1BJqtnOVa0c4iWUA-mpBYGH88zosgY9ba)

All my other songs export just fine. I am only having issues with this specific song.

**Comment by JDRLGaming** [comment]
i am having the same problem with my song when i use custom instruments. its been a problem for over a month

**Comment by Follery** [comment]
It seems to only export fine when you lock enough layers. No layers over 10, unless there are few-enough notes.

**Comment by Bentroen** [comment]
@Follery Funnily enough, I was able to export it without any problem. And believe it or not, I'm pretty sure it has to do with my screen size!

I'm pretty convinced I've found the cause of the crash now. The array that stores whether or not a layer is locked (which is read during the datapack and MP3 export) was only being updated for the layers currently drawn to the screen, so if you have a song with more layers than it fits in your screen and never scroll down to the bottom, the values for these layers will be never set, causing a crash when the program tries to read their values.

I've made a change that hopefully should fix the problem, both in datapack and MP3 export. If it doesn't, this ticket can be reopened in the future if the issue persists.
